[cheeggered] iP

cheeggered opened the pull request:
### DukePro

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level 0

Commit:
Level 1

Commit:
Level 2

Commit:
Level 3

Commit:
Level 3

Commit:
Level 4

Commit:
Level 5

Commit:
Level 5

Commit:
Level 6

Commit:
A-TextUiTesting

Commit:
A-TextUiTesting

Commit:
Level 7

Commit:
Revert "Level 7"

This reverts commit d6fb3174e84110d24a8c1b616cae6cf9d7436694.

Commit:
Implement: save memory to file

Commit:
Amendments to implementation of save List locally

Commit:
Implement load function

Commit:
Add readFile method to load past data

Commit:
Update data file

Commit:
Update data file

Commit:
Implement LocalDate

Commit:
Update Deadline class to store and deal with LocalDate

Commit:
Update Event class to store and deal with LocalDate

Commit:
Update Task class to store and deal with LocalDate

Commit:
Update Todo class to store and deal with LocalDate

Commit:
Add AddCommand class

Commit:
Add Command class

Commit:
Add DeleteCommand class

Commit:
Add ExitCommand class

Commit:
Add InvalidCommand class

Commit:
Add ListCommand class

Commit:
Add MarkCommand class

Commit:
Add UnmarkCommand class

Commit:
Add Parser class

Commit:
Add Storage class

Commit:
Add Ui class

Commit:
Add TaskList class

Commit:
Update Task class

Commit:
Update CheeChat class

Commit:
Update data file

Commit:
Add packages

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

Commit:
Update main class

Commit:
Update data file

Commit:
Update TaskList class to handle IndexOutOfBound for delete method

Commit:
Add empty data file for testing

Commit:
Add StorageTest class to test load() with empty file

Commit:
Add TaskListTest class to test delete() with indexOutOfBound

Commit:
Fix Storage class constructor

Commit:
Update shadowJar

Commit:
Add JavaDoc comments

Commit:
Revert "Add JavaDoc comments"

This reverts commit 76913949e78adc5ce6834cfc1ffee502e915774b.

Commit:
Add JavaDoc comments

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments

Commit:
Add JavaDoc comments for AddCommand class

Commit:
Refactor to comply with the given coding standard

Commit:
Refactor to comply with the given coding standard

Commit:
Update data file

Commit:
Add FindCommmand class

Commit:
Add find command in Parser class

Commit:
Add find method in TaskList class

Commit:
Add printFind method in Ui class to print tasks containing specified keyword

Commit:
Merge branch-Level-9 to master branch

Commit:
Add new files to set up JavaFX

Commit:
Update build.gradle to implement JavaFX

Commit:
Update execute method of Command classes

Commit:
Update methods in Ui class to return String instead of printing

Commit:
Add getTaskDescription method

Commit:
Update data file

Commit:
Update printLine method

Commit:
Update getResponse method

aarontxz commented on src/main/java/duke/CheeChat.java:
```diff
@@ -0,0 +1,61 @@
+        tasks = new TaskList(storage.load());
+    }
+
+    protected String getResponse(String input) {
```
I like that these steps are being packaged as one function that complys with the meaning of name of the function.

aarontxz commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,25 @@
+
+public class FindCommand extends Command {
+
+    private String find;
```
would it be better to name this string keyword instead of find to prevent confusion with the find function in TaskList

aarontxz commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+     * @param num The index number of the Task in the TaskList.
+     * @return The Task that got deleted.
+     */
+    public Task delete(int num) {
```
would it be better to let the user know that they have entered an invalid index instead of returning null? or was this handled somewhere else

aarontxz commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+     * @return An ArrayList containing the tasks containing the keyword.
+     */
+    public ArrayList<Task> find(String keyword) {
+        ArrayList<Task> containsKey = new ArrayList<>();
```
would it be better to name this ArrayList of tasks, tasksContainingkey for it to be more intuitive?

aarontxz reviewed (commented):
Very well written code with good packaging and coding practices, just some minor things to fix.

btaanish commented on src/main/java/duke/CheeChat.java:
```diff
@@ -0,0 +1,61 @@
+    protected String getResponse(String input) {
+        String output = "";
+        try {
+            Command c = Parser.parse(input);
```
Perhaps adding contexual variable names might be more intuitive?

btaanish commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,44 @@
+     * @param storage The Storage instance that will update the file.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
```
Perhaps expressing functionality through the method name might be more explanatory like maybe "executeTask" or "executeAction"?

btaanish commented on src/main/java/duke/CheeException.java:
```diff
@@ -0,0 +1,15 @@
+     * @param message The detail message that describes the reason for this exception.
+     *                It should provide context about the error condition.
+     */
+    public CheeException(String message){
```
I like how javadocs are added for most public functions.

btaanish commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,77 @@
+            return new MarkCommand(number);
+        } else if (command.startsWith("unmark")) {
+            int index = command.length() - 1;
+            char c = command.charAt(index);
```
Perhaps having an expressive variable name might be a better strategy?

btaanish commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,77 @@
+            int index1 = command.indexOf(47);
+            String description = command.substring(6, index1);
+            String duration = command.substring(index1 + 6);
+            int index2 = duration.indexOf(47);
```
Perhaps a better strategy would be to avoid using digits in variable names?

btaanish commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+     * @return The String representation of the TaskList.
+     */
+
+    public String getTaskDescription(int index) {
```
I like how method names here clearly express their functionality.

btaanish reviewed (commented):
Overall well attempted with a clear and close adherence to the coding standars. Maybe a few nits here and there but generally a really well designed program!

Commit:
Add assertions to Storage class

cheeggered committed:
Merge pull request #1 from cheeggered/branch-A-Assertions

Add assertions to Storage class

Commit:
Refactor Parser, Storage and Task class

Methods in these classes had multiple levels of abstraction. To improve code quality and readability.

Let's,
* Add new methods in parse class to process each individual command

Commit:
Refactor Command classes to change method name from execute to executeTask

Commit:
Refactor CheeChat class to change variable names for better clarity

Commit:
Refactor Parser class to change variable names for better clarity

Commit:
Refactor TaskList class to change variable names for better clarity

cheeggered committed:
Merge branch 'master' into branch-A-CodeQuality

cheeggered committed:
Merge pull request #2 from cheeggered/branch-A-CodeQuality

Improve code quality

Commit:
Add priority feature

Priority is added for users to differentiate between high, normal and low priority tasks.

Let's,
* Add Priority field in Task class
* Add PriorityList enum
* Add PriorityCommand class to execute priority commands from users
* Update parse to check for priority commands
* Implement Comparator and Comparable to compare task priorities
* Update Ui to print message when priority is successfully updated

Commit:
Update data file

Commit:
Update to improve GUI

Commit:
Update README.md to include all features added

Commit:
Update Parser class to improve code quality

Commit:
Update data file

Commit:
Upload Product Screenshot

Commit:
Update method in Parser class

Commit:
Update README.md
